R/zchunk_LA115.RooftopPV.R

```diff
@@ -30,43 +28,125 @@ module_gcam.usa_LA115.RooftopPV_DISABLED <- function(command, ...) {
     states_subregions <- get_data(all_data, "gcam-usa/states_subregions")
     NREL_Com_PV_supply_curve <- get_data(all_data, "gcam-usa/NREL_Com_PV_supply_curve")
     NREL_Res_PV_supply_curve <- get_data(all_data, "gcam-usa/NREL_Res_PV_supply_curve")
-    A15.roofPV_TechChange <- get_data(all_data, "energy/A15.roofPV_TechChange")
 
     # ===================================================
-    # TRANSLATED PROCESSING CODE GOES HERE...
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: there are `merge` calls in this code. Be careful!
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: there are 'match' calls in this code. You probably want to use left_join_error_no_match
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # ===================================================
 
-    # Produce outputs
-    # Temporary code below sends back empty data frames marked "don't test"
-    # Note that all precursor names (in `add_precursor`) must be in this chunk's inputs
-    # There's also a `same_precursors_as(x)` you can use
-    # If no precursors (very rare) don't call `add_precursor` at all
-    tibble() %>%
-      add_title("descriptive title of data") %>%
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
+    # Compute PV fixed charge rate
+    PV_FCR <- ( PV.DISCOUNT_RATE * ( 1 + PV.DISCOUNT_RATE ) ^ PV.LIFETIME ) /
+      ( ( PV.DISCOUNT_RATE + 1 ) ^ PV.LIFETIME - 1 )
+
+    # Get maximum residential capacity factor
+    filter(NREL_Res_PV_supply_curve, Relative_Cost == min(Relative_Cost))$MWh /
+      filter(NREL_Res_PV_supply_curve, Relative_Cost == min(Relative_Cost))$MW /
+      HOURS_PER_YEAR -> max_resid_capacity_factor
```

Comment:
This is kind of confusing. What's coming out of this–a vector or a scalar?

Reply:
I second Ben's question.

Reply:
It's just a constant. I'll clarify in the comments.